app/gui/src/presenter/searcher.rs

```diff
@@ -346,6 +346,16 @@ impl Searcher {
 
             action_list_changed <- source::<()>();
             select_entry <- action_list_changed.filter(f_!(model.should_auto_select_first_action()));
+
+            eval_ model.view.toggle_private_component_browser_entries_visibility (model.controller.reload_list());
+            action_list_loaded <-
+                action_list_changed.map(f_!([model] !model.controller.actions().is_loading()));
+            action_list_loaded <- action_list_loaded.on_true();
+            reloading_action_list <-
+                bool(&action_list_loaded, &model.view.toggle_private_component_browser_entries_visibility);
+            reloading_action_list <- reloading_action_list.on_change();
+            reloading_action_list_done <- reloading_action_list.on_false();
+            eval_ reloading_action_list_done (model.controller.update_filtering());
```

Comment:
So, I think `update_filtering` should be called not here in presenter, but in controllers - each time when we set up the list (in `gather_actions_from_engine` just before sending a notification about new list).